Settings: Join theme sections + UI polish (frio)

marcusxss opened the pull request:
Closes #15307 #12265

The two theme sections - theme selection and theme settings are joined into one. One still needs to save changes to see settings for the new theme, though, so added an info text explaining that. (Currently "save changes" is also needed to get the settings for the newly selected theme, so that's unchanged, but perhaps it could be a bit more confusing now they're right next to each other. On the other hand I think it's less confusing that all theme related settings are in the same section).

Highlight colors are selected is changed mostly following the suggestions in #12265.

The description for Vier is updated minimally.

Also did some minor refactoring in main.js by using the more readable template literals over of a lot of appending.

Source strings **are** changed/updated.

# Questions

I haven't tested out when there's a mobile theme set. Even on the backend I don't see an option to enable it. There's just a dropdown with this one option:
<img width="603" height="104" alt="image" src="https://github.com/user-attachments/assets/6f258a96-29c9-4d0d-969d-bd80f17bac89" />
Maybe I'm looking in the wrong place?

# Before

<img width="677" height="637" alt="image" src="https://github.com/user-attachments/assets/b9fdf283-c798-4b28-947a-26215067dc2e" />

<img width="647" height="429" alt="image" src="https://github.com/user-attachments/assets/c0c84e61-dae4-4c68-9953-13e47d474fb7" />

# After

Current version:
<img width="672" height="1017" alt="image" src="https://github.com/user-attachments/assets/27d71a2d-e550-4116-bf34-6cf762fb1d87" />

Earlier versions:
<img width="676" height="1034" alt="image" src="https://github.com/user-attachments/assets/1c8e65ab-6dd7-41b6-ae65-b0d458f36ddf" />

<img width="669" height="1049" alt="image" src="https://github.com/user-attachments/assets/258558ca-0c16-4c0b-a334-acfeb2cc474d" />

marcusxss committed:
Settings: Join theme sections, with UI adjustments (Frio)

marcusxss committed:
Settings: Themes - Move away from the JS approach again

marcusxss committed:
Settings: Theme - Final adjustments

marcusxss commented:
Updated - also the screenshot above - to make the colours smaller, make the outline change between light/dark theme (thanks @foss-) and use a thicker outline to make the current selection stand out further.

foss- commented:
The color dots still look huge to my eyes e.g. in relation to font / titles, checkboxes and other elements. Having them at half or 2/3s the size is probably sufficient.

Should it be `Settings for Frio` (capital letter for theme name)?

marcusxss commented:
They are fairly big, I based the size off your image:
<meta http-equiv="content-type" content="text/html; charset=utf-8"><img src="https://user-images.githubusercontent.com/5708172/203985629-f802133f-4499-4ad9-86ce-31247721a725.png" alt="2" style="max-width: 100%;">

I can reduce their size further. It looks like mine are still a bit bigger than the above.

It's dynamically pulling the theme name that was already available in the file, but I can capitalize it.

foss- commented:
Iirc that was a screenshot from Twitter highlight color selection at that time. It is not the actual size but just a screenshot from a very large monitor. I should have provided more context of the surrounding elements.

marcusxss committed:
Help.php: Also use ucFirst instead of mb_ucfirst as it's a recent PHP feature

marcusxss commented:
Uppercased the theme names with our custom ucFirst, reduced the size of the colours further and changed how they're spaced out on the page.

Now I rediscovered/remembered ucFirst, I also updated my former change to Help.php (docs) to use that instead of `mb_ucfirst`, since everyone may not have a recent enough PHP version for latter.

foss- commented:
Minor typo and changed some wording around. Feel free to adapt as you see fit.

Currently

`Note: If you select a new the theme, you need to save changes before you can see the settings for the newly chosen theme below`

Expected

`Note: If you switch the theme, you need to save changes before you can see the settings for the new theme below.`

marcusxss commented:
Agreed - updated the code (but not the screenshot).

marcusxss committed:
Settings: Theme - Switch to checkmark for choosing accent color

foss- commented:
Wondering if `Themes` is sufficient as headline for this section instead of `Theme Selection`, as it describes what is going on in this part of the settings fairly well and covers more than just the theme selection?

marcusxss commented:
The source string is just "Themes" but then the English language version I'm on (I think English (UK)) "translates" that to "Theme Selection", so that's something that should be fixed on Transifex. (Btw. I think we should probably migrate from Transifex to Weblate as they're open source)

marcusxss commented:
The strangest example I've seen of that, is that someone must've seen "Registration" on the registration page, and decided it was better that it said "Join this Friendica Node today" - and now the Administration settings for registration, which uses the same translation, also says "Join this Friendica Node Today":
<img width="664" height="289" alt="image" src="https://github.com/user-attachments/assets/4f791fb1-4b76-4f29-ab3a-027b6720484a" />

That's in the English (UK) version, which I *have* updated the translations for, though.

foss- commented:
UK English string for Settings `Themes` fixed on transifex. Thanks for this PR. Looking forward to see this trickle down to the instances I am on.

haheute commented:
Maybe, when you select "Vier", the "Settings for Frio", could be hidden.. not displayed.. if possible..
